include/nuttx/bits.h

```diff
@@ -26,46 +26,47 @@
  ****************************************************************************/
 
 #include <assert.h>
-#include <inttypes.h>
-#include <stdint.h>
+#include <limits.h>
 
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 
 #ifndef BITS_PER_BYTE
-# define BITS_PER_BYTE 8
+  #define BITS_PER_BYTE CHAR_BIT
 #endif
 
-#if UINTPTR_MAX > UINT32_MAX
-# define BITS_PER_LONG 64
-#else
-# define BITS_PER_LONG 32
+#ifndef BITS_PER_LONG
+#  define BITS_PER_LONG (sizeof(unsigned long) * BITS_PER_BYTE)
 #endif
 
 #ifndef BITS_PER_LONG_LONG
-# define BITS_PER_LONG_LONG 64
+#  define BITS_PER_LONG_LONG (sizeof(unsigned long long) * BITS_PER_BYTE)
 #endif
 
-#define BIT_MASK(nr)       (UINT32_C(1) << ((nr) % BITS_PER_LONG))
+#define BIT_BYTE_MASK(nr)  (1ul << ((nr) % BITS_PER_BYTE))
+#define BIT_WORD_MASK(nr)  (1ul << ((nr) % BITS_PER_LONG))
+#define BIT_BYTE(nr)       ((nr) / BITS_PER_BYTE)
 #define BIT_WORD(nr)       ((nr) / BITS_PER_LONG)
-#define BIT_ULL_MASK(nr)   (UINT64_C(1) << ((nr) % BITS_PER_LONG_LONG))
+#define BIT_ULL_MASK(nr)   (1ull << ((nr) % BITS_PER_LONG_LONG))
```

Comment:
Why change UINT32_C(1) and UINT64_C(1)?

Reply:
Since UL/ULL is different from UINT32_C/UINT64_C.